src/frontend/http.rs

```diff
@@ -10,77 +16,195 @@ use axum::{
     config::{node_shared_secret, SYS_CFG},
     prelude::*,
 };
 
-async fn write_file_handler(pk: &str, body: BodyStream, name: Option<String>) -> Result<String> {
+// ...
+const SEGMENT_SIZE: usize = 1024 * 1024; // 1 MB
+
+pub async fn write_file_handler(
+    pk: &str,
+    body: BodyStream,
+    name: Option<String>,
+    _url: &Uri,
+) -> Result<String, Error> {
     let pk = &Secp256k1PubKey::try_from(pk)?;
 
-    let file_stream: FileStream = stream::try_unfold(
-        (body, BytesMut::with_capacity(SEGMENT_SIZE * 2)),
-        |(mut body_stream, mut remainder)| async {
-            while remainder.len() < SEGMENT_SIZE {
-                if let Some(chunk) = body_stream.next().await {
-                    let bytes = chunk?;
-                    remainder.extend(bytes);
-
-                    if remainder.len() >= SEGMENT_SIZE {
-                        let segment = remainder.split_to(SEGMENT_SIZE);
-                        trace!("Stream 1MB segment");
-                        return Ok(Some((segment.freeze(), (body_stream, remainder))));
+    let is_mime_type = Arc::new(Mutex::new("attempt_for_mime_type_is?".to_string()));
+
+    let file_stream_result = async {
+        let mut remainder = BytesMut::with_capacity(SEGMENT_SIZE * 2);
+        let mut body_stream = body;
+        let mut buffer = Vec::new();
+        let mut file_data = Vec::new();
+        let mut total_bytes_read = 0;
+
+        while let Some(chunk) = body_stream.next().await {
+            let bytes = match chunk {
+                Ok(bytes) => bytes,
+                Err(_) => todo!(),
+            };
+
+            total_bytes_read += bytes.len();
+            buffer.extend_from_slice(&bytes);
+
+            let kind = infer::get(&buffer);
+            let mime_type = {
+                debug!("INNER Before locking is_mime_type");
+                let mut mime_hasher_guard = is_mime_type.lock().await;
+                debug!("INNER After locking is_mime_type");
+
+                let mut calculated_mime_type = match kind {
+                    Some(mime_type) => mime_type.mime_type().to_string(),
+                    None => "application/octet-stream".to_string(),
+                };
+
+                // Here we check if the inferred type is `text/xml`, which could be an SVG.
+                if calculated_mime_type == "text/xml" {
+                    // Check for the presence of the `<svg` tag in the buffer
+                    // We only read up to a certain number of bytes to check for SVG to prevent having to load a very large file into memory
+                    let svg_check_len = std::cmp::min(buffer.len(), 1024); // Check first 1024 bytes
+                    let content = std::str::from_utf8(&buffer[..svg_check_len]).unwrap_or("");
+                    if content.contains("<svg") {
+                        debug!(">>>>>>>>>>>>>>>> content.contains SVG ");
+
+                        calculated_mime_type = "image/svg+xml".to_string();
```

Comment:
This is neat, but it's still not following my recommendation to use the path extension to detect the type.